documentation/creating-components/tutorial-extending-component.asciidoc

```diff
@@ -6,33 +6,27 @@ layout: page
 
 = Extending Components
 
-Extending a component is an easy way of creating a new component.
+You can create a new component by extending any existing component.
 
-Vaadin flow applications are formed of many web components.
-In most of the cases, those components have a client-side component and a server-side component.
+For most components, there is a client-side component and a corresponding server-side component: 
 
-The client-side component contains the HTML, CSS, and JavaScript.
-This client component defines a set of properties that define the behaviour of the component.
+* *Client-side component*: Contains the HTML, CSS, and JavaScript, and defines a set of properties that determine the behavior of the component on the client side.
 
-The server-side component allows modifying those properties to manage the component's behaviour from Java (server side).
+* *Server-side component*: Contains Java code that allows for changing of the client-side properties, and manages the component behavior on the server side.
 
-== Server Side Extension
+== Extending Server-side Components
 
-Extending a component in flow is useful if it is planned to add a new functionality that does not add any visual aspects to the existing component.
-
-Examples of those new functionalities are automatically processing data, adding default validators, combining multiple simpler components to a field that manages complex data, etc.
+Extending a server-side component is useful when you want to add new functionality (as opposed to visual aspects) to an existing component. Suitable examples include automatically processing data, adding default validators, and combining multiple simple components into a field that manages complex data. 
 
 [WARNING]
-If you plan on adding many new features or changing any visual aspects in the component, it is better to extend the component to the client side.
-Using the Flow Element API excessively creates an extra load in the application.
+If you plan to add many new features or change visual aspects in the component, it is better to extend the component on the client side. Overusing the Flow Element API can create excessive load in the application.
```

Comment:
```suggestion
If  your component contains a lot of logic that could easily be done on the client-side, consider implementing it as a Web Component and creating a wrapper for that. This approach might give you better user experience and less load on the server.
```

Let's make this bit less scary. Reading this might make one think this is some kind of antipattern. @ruthstocks check the sentence I wrote if it makes sense.

Reply:
I edited your suggestion and updated the PR.